lib/generators/boring/pundit/install/install_generator.rb

```diff
@@ -0,0 +1,75 @@
+# frozen_string_literal: true
+
+require 'bundler'
+
+module Boring
+  module Pundit
+    class InstallGenerator < Rails::Generators::Base
+      desc "Adds pundit to the application"
+
+      def add_pundit_gem
+        say "Adding Pundit gem", :green
+
+        Bundler.with_unbundled_env do
+          run "bundle add pundit"
+        end
+      end
+
+      def run_pundit_generator
+        return if options[:skip_generator]
+
+        say "Running Pundit Generator", :green
+        run "DISABLE_SPRING=1 rails generate pundit:install"
+      end
+
+      def inject_pundit_to_controller
+        say "Adding Pundit module into ApplicationController", :green
+        inject_into_file 'app/controllers/application_controller.rb', after: "class ApplicationController < ActionController::Base\n" do
+          "  include Pundit\n"
+        end
+      end
+
+      def ensure_policies
+        return if options[:skip_ensuring_policies]
+
+        say "Force ensuring policies", :green
+        inject_into_file 'app/controllers/application_controller.rb', after: "include Pundit\n" do
+          "  after_action :verify_authorized\n"
+        end
+      end
+
+      def rescue_from_not_authorized
+        return if options[:skip_rescue]
+
+        say "Add rescue from Pundit::NotAuthorizedError", :green
+
+        after = if File.read('app/controllers/application_controller.rb') =~ (/:verify_authorized/)
+                  "after_action :verify_authorized\n"
+                else
+                  "include Pundit\n"
+                end
+
+        inject_into_file 'app/controllers/application_controller.rb', after: after do
+          <<~RUBY
+            rescue_from Pundit::NotAuthorizedError, with: :user_not_authorized
+
+            private
+
+            def user_not_authorized
+              flash[:alert] = "You are not authorized to perform this action."
+              redirect_to(request.referrer || root_path)
+            end
+          RUBY
+        end
+      end
+
+      def after_run
```

Comment:
Yeah, but... A user should know what the installer will add and what everything means and do. I would not recommend anyone, who never saw Pundit, to use the installer. 

I think that everyone should at least once install it step by step manually using the documentation. Only by this, he will know, what everything does and why it is there.

If a user without that knowledge will use it, it will not help him, but only bring him more troubles in the future.

So... link to documentation and issues would be, by me, much better. 

There are a lot of blog posts with "how to install X" but it is up to you...

Reply:
Thank you. I understand the concern and share the same opinion 👍